syntax = "proto3";

import "google/protobuf/empty.proto";
import "google/protobuf/timestamp.proto";

service OrderService {
  rpc GetOrders(google.protobuf.Empty) returns (Orders);
  rpc CreateOrder(Order) returns (Order);
  rpc GetOrder(OrderID) returns (Order);
  rpc CancelOrder(OrderID) returns (google.protobuf.Empty);
  rpc DeleteOrder(OrderID) returns (google.protobuf.Empty);
}

message OrderID {
  string id = 1;
}

message Item {
  string product = 1;
  int32 quantity = 2;
}

message Order {
  string id = 1;
  repeated Item items = 2;
  string status = 3;
  google.protobuf.Timestamp created = 4;
  string customer_id = 5;
  string kitchen_id = 6;
  string delivery_id = 7;
}

message Orders {
  repeated Order orders = 1;
}